src/Hl7.Fhir.Base/ElementModel/PocoElementNode.cs

```diff
@@ -67,50 +71,89 @@ private Type determineInstanceType(PropertyMapping definition)
                 "url" => typeof(FhirUri),
                 "id" => typeof(FhirString),
                 "div" => typeof(XHtml),
-                _ => throw new NotSupportedException($"Encountered unexpected primitive type {Name} in backward compat behaviour for PocoElementNode.InstanceType.")
+                _ => throw new NotSupportedException(
+                    $"Encountered unexpected primitive type {Name} in backward compat behaviour for PocoElementNode.InstanceType.")
             };
         }
 
         public IElementDefinitionSummary Definition { get; private set; }
 
         public string ShortPath { get; private set; }
 
-        public IEnumerable<ITypedElement> Children(string name)
+        private (ElementValue val, int ix)[] returnElements(string name, object element)
         {
-            if (!(Current is Base parentBase)) yield break;
+            return (element, Current, name) switch
+            {
+                (Base @base, _, _)  => new[] { (new ElementValue(name, @base), 0) },
+                (IEnumerable<Base> bases, _, _)  => bases.Select((e, i) => (new ElementValue(name, e), i)).ToArray(),
+                (string s, Extension, "url")  => new[]{(new ElementValue(name, new FhirUri(s)), 0)},
+                (string s, Element, "id")  => new[]{(new ElementValue(name, new FhirString(s)), 0)},
+                _ => Array.Empty<(ElementValue, int)>(),
+            };
+        }
 
-            var children = parentBase.NamedChildren;
-            string oldElementName = null;
-            int arrayIndex = 0;
+        public IEnumerable<ITypedElement> Children(string name)
```

Comment:
The crucial difference is that previously:
- it got *all* children with `NamedChildren`
- then visited that whole list until all the items with the right name were seen
- and maintained the array index per child name along the way
And currently:
- it utilizes IReadOnlyDictionary to quickly get *only* the children with the correct name
- appends the array index

I see your point that ElementValue may not be necessary anymore, since `ElementValue.name` is not even used. I will check that.

Reply:
ElementValue was indeed not needed anymore, I removed it from this method.
Note that NamedChildren and ElementValue have no use at all anymore now in the SDK itself. But they are public...

Reply:
Yes, and I think e.g. Forge is using it, since it was added by Michel initially!